src/main/java/com/laborete/LaboreteAPI/profile/services/UsersServiceImpl.java

```diff
@@ -102,38 +129,127 @@ public ResponseEntity<HttpStatus> uploadUserAvatar(String name, MultipartFile fi
+
+            UserAvatarEntity userAvatarEntity = new UserAvatarEntity(
+                    uuid,
+                    name,
+                    file.getSize(),
+                    FileUtils.getFileExtension(file),
+                    fileTypeMap.getContentType(avatar));
             user.setUserAvatarEntity(userAvatarEntity);
             this.userAvatarRepository.save(userAvatarEntity);
             this.usersRepository.save(user);
+            usersRepository.flush();
+
 
             return new ResponseEntity<>(HttpStatus.CREATED);
         } catch (Exception e) {
             throw new ResourceFileUploadErrorException(FILE_FAILED_UPLOAD, e);
         }
     }
 
-    private String createDirectory(UUID uuid) {
+    @Transactional
+    public ResponseEntity<HttpStatus> uploadUserBackground(String name, MultipartFile file, UUID userId) {
+        UserEntity user = this.usersRepository.getUserById(userId)
+                .orElseThrow(
+                        () -> new ResourceNotFoundException(USER_NOT_FOUND + userId)
+                );
+        if (file.isEmpty()) {
+            throw new ResourceBadRequestException(FILE_IS_EMPTY);
+        }
+        if (name.isBlank() || name.isEmpty()) {
+            throw new ResourceBadRequestException(FILE_NAME_IS_EMPTY);
+        }
+
         try {
-            File directory = new File(Objects.requireNonNull(ROOT_PATH));
+            byte[] bytes = file.getBytes();
+
+
+            UUID uuid = UUID.randomUUID();
+            var path = createDirectory(uuid, ROOT_PATH_BACKGROUND);
+            MimetypesFileTypeMap fileTypeMap = new MimetypesFileTypeMap();
+
+            File background = new File(path, uuid + FileUtils.getFileExtension(file));
+            BufferedOutputStream stream = new BufferedOutputStream(new FileOutputStream(background));
+
+            stream.write(bytes);
+            stream.close();
+
+            UserBackgroundEntity userBackgroundEntity = new UserBackgroundEntity(
+                    uuid,
+                    name,
+                    file.getSize(),
+                    FileUtils.getFileExtension(file),
+                    fileTypeMap.getContentType(background));
+
+            user.setUserBackgroundEntity(userBackgroundEntity);
+            userBackgroundRepository.save(userBackgroundEntity);
+            usersRepository.save(user);
+            usersRepository.flush();
+
+
+            return new ResponseEntity<>(HttpStatus.CREATED);
+
+        } catch (Exception e) {
+            throw new ResourceFileUploadErrorException(FILE_FAILED_UPLOAD, e);
+        }
+    }
+
+    private byte[] getBytesByIdFromAvatar(UUID uuid) {
+        UserAvatarEntity userAvatarEntity = userAvatarRepository.getUserAvatarById(
+                uuid).orElseThrow(
+                () -> new ResourceNotFoundException(AVATAR_NOT_FOUND + uuid)
+        );
+        String pathVar = getDirectory(uuid, ROOT_PATH_AVATARS) + "\\" + uuid + userAvatarEntity.getExtension();
```

Comment:
Code smell

Reply:
I need more additional explanations about it